### PR TITLE
Permissive slugs, Refs# 11761

### PR DESCRIPTION
**Pull request by sbreker**
Add setting to Admin page to trigger new slug generation.

### CODE REVIEW

File: lib/task/migrate/migrations/arMigration0159.class.php [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Access to Memory (AtoM).  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/*
+ * Add new setting for enabling institutional scoping.
```
Comment (jraddaoui) [inline lib/task/migrate/migrations/arMigration0159.class.php]: Comment from other migration

File: lib/task/migrate/migrations/arMigration0159.class.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  public function up($configuration)
+  {
+    if (null === QubitSetting::getByName('permissive_slug_creation'))
+    {
+      $setting = new QubitSetting;
+      $setting->name = 'permissive_slug_creation';
+      $setting->value = QubitSlug::SLUG_RESTRICTIVE;
+      $setting->editable = 1;
+      $setting->culture = 'en';
+      $setting->save();
```
Comment (jraddaoui) [inline lib/task/migrate/migrations/arMigration0159.class.php]: Just to be sure the value is added to the source culture, I prefer:

```
$setting->source_culture = 'en';
$setting->setValue(QubitSlug::SLUG_RESTRICTIVE, array('culture' => 'en'));
```

File: lib/task/migrate/migrations/arMigration0159.class.php [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+
+    $sql = "ALTER TABLE `slug` MODIFY `slug` VARCHAR(255) CHARACTER SET utf8 COLLATE utf8_bin";
```
Comment (jraddaoui) [inline lib/task/migrate/migrations/arMigration0159.class.php]: Do we need to modify the SQL schema to create the database with this change in new installs? Also, does it require to rebuild the models?

Comment (jraddaoui) [reply lib/task/migrate/migrations/arMigration0159.class.php]: Oh, I see it now. Nicely done ;)

File: plugins/sfInstallPlugin/lib/sfInstall.class.php [hunk 4]
```diff
@@ -531,4 +533,17 @@ public static function addSymlinks()
 
     return $symlinks;
   }
+
+  /**
+   * Make any additional sql changes that cannot be made via schema.yml.
+   * Needs to be run for new installs and the tools:purge task.
```
Comment (jraddaoui) [inline plugins/sfInstallPlugin/lib/sfInstall.class.php:539]: Should it be called from that task then?

File: test/unit/slugifyTest.php [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  $t->is(QubitSlug::slugify($given, QubitSlug::SLUG_RESTRICTIVE), $expected, "slugify(SLUG_RESTRICTIVE) creates slug from text: $given as $expected");
+}
+
+// Test Permissive mode slug creation.
+foreach ($testsPermissive as $item)
+{
+  list($given, $expected) = $item;
+
+  $t->is(QubitSlug::slugify($given, QubitSlug::SLUG_PERMISSIVE), $expected, "slugify(SLUG_PERMISSIVE) creates slug from text: $given as $expected");
+}
```
Comment (jraddaoui) [inline test/unit/slugifyTest.php:72]: Awesome!

**Review by jraddaoui — APPROVED** [review]
Looks great! Just a couple of minor comments.

File: lib/task/migrate/migrations/arMigration0159.class.php (hunk 1 above)
Comment (sbreker) [reply lib/task/migrate/migrations/arMigration0159.class.php]: You are right!  Will take this out...

File: lib/task/migrate/migrations/arMigration0159.class.php (hunk 3 above)
Comment (sbreker) [reply lib/task/migrate/migrations/arMigration0159.class.php]: Our version of Propel has a bug where I cannot set individual column collation.  Was forced to do that in sfInstall plugin for new installs and the CLI purge task instead.

File: plugins/sfInstallPlugin/lib/sfInstall.class.php (hunk 4 above)
Comment (sbreker) [reply plugins/sfInstallPlugin/lib/sfInstall.class.php:539]: sfInstall::loadData() is called from the tools:purge task here:

https://github.com/artefactual/atom/blob/qa/2.5.x/lib/task/tools/purgeTask.class.php#L115

File: test/unit/slugifyTest.php (hunk 5 above)
Comment (sbreker) [reply test/unit/slugifyTest.php:72]: Thanks!  :)

**Comment by jraddaoui** [comment]
Thanks! Please, merge in one single commit. And I think it looks better with refs in lowercase, as it's after a comma an with the # in the issue number.

**Comment by jraddaoui** [comment]
Merged in 648d15a

**Pull request closed without merging**